tests/PersistentArrayTest.php

```diff
@@ -232,6 +232,52 @@ public function testTryLoadNotSupportedException()
         $this->expectException(Exception::class);
         $m->tryLoad(1);
     }
+    
+    /**
+     * different to other field types, date, time and datetime fields return an
+     * object, not a scalar value to work with. Make sure Array persistence handles them correctly.
+     */
+    public function testDateTimeFields() {
+        $p = new Persistence\Array_();
+        $m = new Model($p);
+        $m->addField('date', ['type' => 'date']);
+        $m->addField('time', ['type' => 'time']);
+        $m->addField('datetime', ['type' => 'datetime']);
+        $m->save();
+
+        //try setting DateTime Object
+        $dateTime = \DateTime::createFromFormat('Y-m-d H:i:s', '2020-01-01 11:11:11');
+        $m->set('date', $dateTime);
+        $m->set('time', $dateTime);
+        $m->set('datetime', $dateTime);
+        $m->save();
+
+        self::assertInstanceOf(
```

Comment:
keep on 1 line

Reply:
Does StyleCI command this? I find assertions way more readable this way.

Reply:
1 line - keep files short - important in general

`self::`, or better `static::` should be more correct, but refactor as another PR everywhere, AFAK there is CS fixer rule for it